Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -703,7 +707,6 @@ void UpdateXRView(MTKView* activeXRView) {
                     MTLTextureDescriptor *textureDescriptor = [MTLTextureDescriptor texture2DDescriptorWithPixelFormat:MTLPixelFormatBGRA8Unorm width:width height:height mipmapped:NO];
                     textureDescriptor.usage = MTLTextureUsageRenderTarget | MTLTextureUsageShaderRead;
                     id<MTLTexture> texture = [metalDevice newTextureWithDescriptor:textureDescriptor];
-                    [texture retain];
```

Comment:
Is there no danger that this could be purged from underneath us while it's still actively being used from the void* pointer below?

Could we alternatively just add cleaning up the ColorTexturePointer to the destructor?

Reply:
<img width="946" alt="image" src="https://user-images.githubusercontent.com/79382720/119389505-91a0c380-bc80-11eb-8199-bfedb8aa443b.png">

**Old approach** : That second line is the extra retain we were doing.
If we weren't keeping a reference at the class level [i.e. std::vector<Frame::View> ActiveFrameViews{ {} };], it would be purged.

Reply:
In the screenshot above, it looks to me like the ref count goes to 1 when the object is created (`newTextureWithDescriptor`), goes to 2 when we explicitly call `retain`, and then goes back to 1 when the `id<MTLTexture>` goes out of scope. If you remove the `retain`, then isn't the ref count going to 0 is soon as `id<MTLTexture>` goes out of scope?

Reply:
Realized that the above screenshot was for another object (shader library).

Reply:
We found that the object here created with new (newTextureWithDescriptor) is retained and not auto released. The explicit retain is not required and needs to be explicitly released (done in destructor).